spec/features/admin/bulk_product_update_spec.rb

```diff
@@ -281,7 +281,7 @@
     expect(p.variant_unit).to eq "weight"
     expect(p.variant_unit_scale).to eq 1000 # Kg
     expect(p.available_on).to eq 3.days.ago.beginning_of_day
-    expect(p.primary_taxon).to eq t1
+    expect(p.primary_taxon.permalink).to eq "ruby-on-rails"
```

Comment:
does category/primary taxon influence permalink? or are you changing the test and not validating the taxon has changed as expected? I think in that case line 270 above should be deleted.
It would be better to keep the taxon validation in some way...